features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/RequestBodyFactory.kt

```diff
@@ -18,78 +19,45 @@ internal class RequestBodyFactory(
-                RAW_SEGMENT_SIZE_FORM_KEY,
-                compressedData.size.toString()
-            )
-            .addFormDataPart(
-                START_TIMESTAMP_FORM_KEY,
-                segment.start.toString()
-            )
-            .addFormDataPart(
-                END_TIMESTAMP_FORM_KEY,
-                segment.end.toString()
-            )
-            .addFormDataPart(
-                SOURCE_FORM_KEY,
-                segment.source.toJson().asString
-            )
-            .build()
+    @Suppress("UnsafeThirdPartyFunctionCall")
+    private fun buildRequestBody(segments: List<Pair<MobileSegment, JsonObject>>): RequestBody {
+        val multipartBody = MultipartBody.Builder().setType(MultipartBody.FORM)
+        val metadata = JsonArray()
+        segments.forEachIndexed { index, segment ->
+            val segmentAsByteArray = (segment.second.toString() + "\n").toByteArray()
+            val compressedData = compressor.compressBytes(segmentAsByteArray)
+            val segmentAsJson = segment.first.toJson().asJsonObject.apply {
+                addProperty(COMPRESSED_SEGMENT_SIZE_FORM_KEY, compressedData.size)
+                addProperty(RAW_SEGMENT_SIZE_FORM_KEY, segmentAsByteArray.size)
+            }
+            multipartBody.addFormDataPart(
+                    name = SEGMENT_DATA_FORM_KEY,
+                    filename = "${BINARY_FILENAME_PREFIX}$index",
+                    body = compressedData.toRequestBody(CONTENT_TYPE_BINARY.toMediaTypeOrNull())
+            )
+            metadata.add(segmentAsJson)
+        }
+        multipartBody.addFormDataPart(
+               name = EVENT_NAME_FORM_KEY,
+                filename = BLOB_FILENAME,
+                body = metadata.toString().toRequestBody(CONTENT_TYPE_JSON.toMediaTypeOrNull())
+        )
+        return multipartBody.build()
     }
 
     companion object {
 
-        internal const val APPLICATION_ID_FORM_KEY = "application.id"
-        internal const val SESSION_ID_FORM_KEY = "session.id"
-        internal const val VIEW_ID_FORM_KEY = "view.id"
-        internal const val HAS_FULL_SNAPSHOT_FORM_KEY = "has_full_snapshot"
-        internal const val RECORDS_COUNT_FORM_KEY = "records_count"
+        internal const val BINARY_FILENAME_PREFIX = "file"
+        internal const val BLOB_FILENAME = "blob"
+        internal const val EVENT_NAME_FORM_KEY = "event"
         internal const val RAW_SEGMENT_SIZE_FORM_KEY = "raw_segment_size"
-        internal const val START_TIMESTAMP_FORM_KEY = "start"
-        internal const val END_TIMESTAMP_FORM_KEY = "end"
-        internal const val SOURCE_FORM_KEY = "source"
-        internal const val SEGMENT_FORM_KEY = "segment"
+        internal const val COMPRESSED_SEGMENT_SIZE_FORM_KEY = "compressed_segment_size"
+        internal const val SEGMENT_DATA_FORM_KEY = "segment"
         internal const val CONTENT_TYPE_BINARY = "application/octet-stream"
+        internal const val CONTENT_TYPE_JSON = "application/json"
```

Comment:
It makes sense to change them from `String` to `MediaType`, this will remove repetitive `toMediaTypeOrNull` calls for the static input.

Reply:
Honestly I did not find any like that in the `MediaType` class, I am going to look more. Are you sure they exist ? Can you give me a link ?

Reply:
I mean you can have `internal val CONTENT_TYPE_JSON = "application/json".toMediaTypeOrNull()`